.gitignore

```diff
@@ -2,6 +2,7 @@
 .DS_Store
 /node_modules
 /*.tgz
+.idea
```

Comment:
This belongs in your local global `.gitignore` instead

Reply:
For all projects in my company, some files are committed from the `.idea` folder to the Git repository. This is done to enforce company-wide styling and IDE configurations. So this is not a viable solution for developers using Webstorm.

Reply:
Just a heads up but this is an open source project, not a project in your company. Typical open source avoids having every contributor adding their custom editor configurations into `.gitignore` over time, and instead asks you as a contributor to not commit irrelevant files. 

> So this is not a viable solution for developers using Webstorm.

I'm not clear what this means? It's not viable to not commit a file? What about people using sublime text, vs code, etc?

Reply:
> Typically open source avoids having every contributor adding their custom editor configurations

yes, that is correct, but in my opinion it is better to add a comprehensive `.gitignore`. This contains the recommended global ignore rules. This is especially useful in open source projects since there is no common global `.gitignore`. And there is no way to specify a global `.gitignore` for public open source project.

> The easiest way to do that is add your own global .gitignore file locally, which you can do once instead of for every project you contribute to. GitHub has a tutorial here.

The problem is that I can't do that. This would lead to the problem that if I am working on a project that requires some webstorm configurations to be committed. The webstorm configuration files are not tracked by Git.

Reply:
I think you can add `!.idea` to your company projects to have them explicitly opt into those folders/files

Reply:
I think this would not work. Some of the files and folders in the `.idea` folder should be ignored. I don't know in detail how the modifier `!` works with ignore in subfolders.

```
!.idea/
.idea/workspace.xml
```

Would this `.gitignore` ignore the file `.idea/workspace.xml`? 

---

This would also be quite a time consuming task. All our dozens of projects would have to be updated, the dozens of template projects and the documentation.

Reply:
I think it would respect the order in which its given, but can't say for certain.

But yeah, I understand your pain, have encountered similar situations in the past :/